lib/media/content_workarounds.js

```diff
@@ -149,6 +150,110 @@ shaka.media.ContentWorkarounds = class {
     return modifiedInitSegment;
   }
 
+  /**
+   * @param {!BufferSource} mediaSegmentBuffer
+   * @return {!Uint8Array}
+   */
+  static fakeMediaEncryption(mediaSegmentBuffer) {
+    const mediaSegment = shaka.util.BufferUtils.toUint8(mediaSegmentBuffer);
+    const mdatBoxes = [];
+    new shaka.util.Mp4Parser()
+        .box('mdat', (box) => {
+          mdatBoxes.push(box);
+        })
+        .parse(mediaSegment);
+
+    const newSegmentChunks = [];
+    for (let i = 0; i < mdatBoxes.length; i++) {
+      const prevMdat = mdatBoxes[i - 1];
+      const currMdat = mdatBoxes[i];
+      const chunkStart = prevMdat ? prevMdat.start + prevMdat.size : 0;
+      const chunkEnd = currMdat.start + currMdat.size;
+      const chunk = mediaSegment.subarray(chunkStart, chunkEnd);
+      newSegmentChunks.push(
+          shaka.media.ContentWorkarounds.fakeMediaEncryptionInChunk_(chunk));
+    }
+    return shaka.util.Uint8ArrayUtils.concat(...newSegmentChunks);
+  }
+
+  /**
+   * @param {!Uint8Array} mediaSegment
+   * @return {!Uint8Array}
+   */
+  static fakeMediaEncryptionInChunk_(mediaSegment) {
```

Comment:
Might be nice to rename `mediaSegment` to something like `chunk` or `moof` in here, since we aren't getting a full media segment anymore.

Reply:
Done!